tests/unit/amo/utils/test_server.js

```diff
@@ -81,4 +84,52 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('viewHeartbeatHandler', () => {
+    // eslint-disable-next-line jest/no-done-callback
+    it('exposes the version.json file', async (done) => {
+      const apiHost = 'https://somehost/';
+      const apiPath = 'some/path/';
+      const apiVersion = 'someVersion';
+      const siteStatus = { read_only: false, notice: 'some notice' };
+      const _config = getFakeConfig({ apiHost, apiPath, apiVersion });
+      const _fetch = sinon
+        .stub()
+        .resolves({ json: sinon.stub().resolves(siteStatus), status: 200 });
+      const handler = viewHeartbeatHandler({ _config, _fetch });
+
+      const res = new MockExpressResponse();
+      handler(null, res);
+
+      res.on('finish', () => {
+        // console.log(_fetch.getCalls());
+        // expect(_fetch).toHaveBeenCalledWith(
```

Comment:
For some reason I cannot get this assertion to work. I kept the `console.log` above, because running the test with it shows that `_fetch` was in fact called with the expected argument, but for some reason the `expect` is throwing an error: 
```
    expect(received).toHaveBeenCalledWith(...expected)

    Matcher error: received value must be a mock or spy function

    Received has type:  function
    Received has value: [Function functionStub]
```

I assume it has something to do with the async nature of the function, but I can't seem to figure out how to resolve this. I would like to include this assertion though.

Reply:
I think we should `await handler(...);` and we might not need `res.on('finish', () => {})`.

Reply:
You are correct that we can eliminate `res.on('finish'...` by using `await handler`, but it doesn't solve the above error, which is still reported when I try to do the `expect`.

Reply:
ah yes, please use `jest.fn()` instead of `sinon`: you're currently using a sinon mock with a jest expecter/matcher.

Reply:
Hmm, I thought I tried that too, but I'll try again.